csi.proto

```diff
@@ -172,6 +189,36 @@ message CreateVolumeRequest {
   // logged, etc.) by the CO.
   // This field is OPTIONAL.
   map<string, string> controller_create_secrets = 5;
+
+  // Contains all attributes of the source snapshot to create volume
+  // from. This field is OPTIONAL.
+  SourceSnapshot source_snapshot = 6;
```

Comment:
What do you think about using a `oneof` instead and calling this something like `ContentSource` where one of the possibilities is a snapshot reference?